channeldb/db.go

```diff
@@ -534,6 +535,38 @@ func fetchChannels(d *DB, pending, waitingClose bool) ([]*OpenChannel, error) {
 	return channels, nil
 }
 
+// FetchPrivateOpenChannelIDs will return a map with the IDs of all open
+// private channels as keys, which can be used for checking if an edge in
+// the graph is private or not.
+// IDs of both 'default' open channels and 'waiting to be closed' private
+// channels are returned. IDs of pending opened channels are not returned,
+// as those are not yet advertised to the graph.
+func (d *DB) FetchPrivateOpenChannelIDs() (map[uint64]bool, error) {
```

Comment:
Would be awesome if this could be added in its own commit, as it is independent from the RPC change. However, I'm still not convinced we need this method, and cannot just rely on the `authProof`.